src/Type/Pretty.hs

```diff
@@ -192,8 +192,9 @@ ppSchemeEffect env tp
 
 prettyDefFunType :: Env -> [ParamInfo] -> Scheme -> Doc
 prettyDefFunType env pinfos tp 
-  = let (Just params,pre,post) = ppDeclType env pinfos tp
-    in pre <.> parens (commaSep (map ppParam params)) <+> text "->" <+> post
+  = case ppDeclType env pinfos tp of
+      (Just params,pre,post) -> pre <.> parens (commaSep (map ppParam params)) <+> text "->" <+> post
+      (Nothing,pre,post) -> pre <+> text "()" <+> text "->" <+> post
```

Comment:
The nothing case does happen occasionally, when showing function types on hover.